front/pages/api/w/[wId]/vaults/index.ts

```diff
@@ -31,7 +31,25 @@ async function handler(
 
   switch (req.method) {
     case "GET":
-      const vaults = await VaultResource.listWorkspaceVaults(auth);
+      const { role } = req.query;
```

Comment:
Should we check first that the user is admin if role is admin to return a proper authorization error?

Reply:
I didn't want to do that initially to avoid conditionnal code in the client but I will change.